Lab3/snake.js

```diff
@@ -623,16 +620,22 @@ const settings = {
 
     snakeType : function(getValue) {
         // TODO: return the Type selected by the user
+        const el = document.getElementById("snake-type")
         if (getValue) {
-            return document.getElementById("snake-type").value;
+            return el.value;
         }
         else {
-            return document.getElementById("snake-type").options[document.getElementById("snake-type").selectedIndex].text;
+            if ( el.value ) {
+                return el.options[el.selectedIndex].text;
+            } else {
+                return "";
+            }
```

Comment:
Protip: if you find yourself writing things like `document.getElementById("snake-type")` multiple times in one place, assign it to a variable and save yourself some typing!
The additional if statement here is to prevent the error that occurs when the user had not yet selected a Type, in which case `.options[selectedIndex]` is null, so you can't access the `.text` property